src/lib/storage.ts

```diff
@@ -20,33 +21,71 @@ export default class AppStorage<Storage extends Record<string, any>/* Storage ex
   //private cache: Partial<{[key: string]: Storage[typeof key]}> = {};
   private cache: Partial<Storage> = {};
   private useStorage = true;
+
+  private getPromises: Map<keyof Storage, CancellablePromise<Storage[keyof Storage]>> = new Map();
+  private getThrottled: () => void;
+
   private keysToSet: Set<keyof Storage> = new Set();
   private saveThrottled: () => void;
+  private saveResolve: () => void;
 
   constructor(storageOptions: Omit<IDBOptions, 'storeName' | 'stores'> & {stores?: DatabaseStore[], storeName: DatabaseStoreName}) {
     this.storage = new IDBStorage(storageOptions);
 
     AppStorage.STORAGES.push(this);
 
     this.saveThrottled = throttle(async() => {
-      if(!this.keysToSet.size) {
-        return;
+      if(this.keysToSet.size) {
+        const keys = Array.from(this.keysToSet.values()) as string[];
+        this.keysToSet.clear();
+
+        try {
+          //console.log('setItem: will set', key/* , value */);
+          //await this.cacheStorage.delete(key); // * try to prevent memory leak in Chrome leading to 'Unexpected internal error.'
+          //await this.storage.save(key, new Response(value, {headers: {'Content-Type': 'application/json'}}));
+          await this.storage.save(keys, keys.map(key => this.cache[key]));
+          //console.log('setItem: have set', key/* , value */);
+        } catch(e) {
+          //this.useCS = false;
+          console.error('[AS]: set error:', e, keys/* , value */);
+        }
       }
 
-      const keys = Array.from(this.keysToSet.values()) as string[];
-      this.keysToSet.clear();
-
-      try {
-        //console.log('setItem: will set', key/* , value */);
-        //await this.cacheStorage.delete(key); // * try to prevent memory leak in Chrome leading to 'Unexpected internal error.'
-        //await this.storage.save(key, new Response(value, {headers: {'Content-Type': 'application/json'}}));
-        await this.storage.save(keys, keys.map(key => this.cache[key]));
-        //console.log('setItem: have set', key/* , value */);
-      } catch(e) {
-        //this.useCS = false;
-        console.error('[AS]: set error:', e, keys/* , value */);
+      if(this.saveResolve) {
+        this.saveResolve();
+        this.saveResolve = undefined;
       }
-    }, 50, false);
+    }, 16, false);
+
+    this.getThrottled = throttle(async() => {
+      const keys = Array.from(this.getPromises.keys());
+
+      this.storage.get(keys as string[]).then(values => {
```

Comment:
Codacy has a fix for the issue: [Expected parentheses around arrow function argument.](https://app.codacy.com/gh/MikeOwino/tweb/pullRequest?prid=7329411)
```suggestion
      this.storage.get(keys as string[]).then((values) => {
```